cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/monitor/sampling/SamplingUtils.java

```diff
@@ -346,124 +313,70 @@ private static boolean skipBuildingBrokerMetricSample(BrokerLoad brokerLoad, int
+    // Get bootstrap servers
+    String bootstrapServers = (String) configs.get(METRIC_REPORTER_SAMPLER_BOOTSTRAP_SERVERS);
+    if (bootstrapServers == null) {
+      bootstrapServers = bootstrapServers(configs);
     }
 
-    NewTopic newTopic = new NewTopic(topic, partitionCount, replicationFactor);
-    Map<String, String> config = new HashMap<>(2);
-    config.put(RetentionMsProp(), Long.toString(retentionMs));
-    config.put(CleanupPolicyProp(), DEFAULT_CLEANUP_POLICY);
-    newTopic.configs(config);
-
-    return newTopic;
+    // Create consumer
+    long randomToken = RANDOM.nextLong();
+    Properties consumerProps = new Properties();
+    consumerProps.putAll(configs);
+    consumerProps.setProperty(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, bootstrapServers);
+    consumerProps.setProperty(ConsumerConfig.CLIENT_ID_CONFIG, clientIdPrefix + "-consumer-" + randomToken);
+    consumerProps.setProperty(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "latest");
+    consumerProps.setProperty(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG, "false");
+    consumerProps.setProperty(ConsumerConfig.MAX_POLL_RECORDS_CONFIG, Integer.toString(Integer.MAX_VALUE));
+    consumerProps.setProperty(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class.getName());
+    consumerProps.setProperty(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, MetricSerde.class.getName());
+    consumerProps.setProperty(ConsumerConfig.RECONNECT_BACKOFF_MS_CONFIG, configs.get(RECONNECT_BACKOFF_MS_CONFIG).toString());
+    return new KafkaConsumer<>(consumerProps);
   }
 
-  /**
-   * Add config altering operations to the given configs to alter for configs that differ between current and desired.
-   *
-   * @param configsToAlter A set of config altering operations to be populated.
-   * @param desiredConfig Desired config value by name.
-   * @param currentConfig Current config.
-   */
-  private static void maybeUpdateConfig(Set<AlterConfigOp> configsToAlter, Map<String, String> desiredConfig, Config currentConfig) {
-    for (Map.Entry<String, String> entry : desiredConfig.entrySet()) {
-      String configName = entry.getKey();
-      String targetConfigValue = entry.getValue();
-      ConfigEntry currentConfigEntry = currentConfig.get(configName);
-      if (currentConfigEntry == null || !currentConfigEntry.value().equals(targetConfigValue)) {
-        configsToAlter.add(new AlterConfigOp(new ConfigEntry(configName, targetConfigValue), AlterConfigOp.OpType.SET));
-      }
+  private static String bootstrapServers(Map<String, ?> configs) {
+    String bootstrapServers = configs.get(MonitorConfig.BOOTSTRAP_SERVERS_CONFIG).toString();
+    // Trim the brackets in List's String representation.
+    if (bootstrapServers.length() > 2) {
+      bootstrapServers = bootstrapServers.substring(1, bootstrapServers.length() - 1);
     }
+    return bootstrapServers;
```

Comment:
Is this duplicated elsewhere?

Reply:
Moved the bootstrapServers extraction to a util function.